packages/compiler/core/binder.ts

```diff
@@ -267,6 +267,14 @@ export function createBinder(program: Program, options: BinderOptions = {}): Bin
 
   function bindEnumStatement(node: EnumStatementNode) {
     declareSymbol(getContainingSymbolTable(), node, node.id.sv);
+    node.locals = createSymbolTable();
+    for (const enumMember of node.members) {
```

Comment:
This is a good first step but if we add enum members dynamically we won't get a symbol for it. You could imagine doing late-bound symbols for this, initializing the bindings in the symbol table when the enum type is instantiated. Think that's worth doing here?